sdl_android/src/androidTest/java/com/smartdevicelink/test/api/audio/AudioStreamManagerTest.java

```diff
@@ -0,0 +1,313 @@
+        try {
+            Method onOutputFormatChangedMethod = BaseAudioDecoder.class.getDeclaredMethod("onOutputFormatChanged", MediaFormat.class);
+
+            Field outputChannelCountField = BaseAudioDecoder.class.getDeclaredField("outputChannelCount");
+            Field outputSampleRateField = BaseAudioDecoder.class.getDeclaredField("outputSampleRate");
+            Field outputSampleTypeField = BaseAudioDecoder.class.getDeclaredField("outputSampleType");
+
+            onOutputFormatChangedMethod.setAccessible(true);
+            outputChannelCountField.setAccessible(true);
+            outputSampleRateField.setAccessible(true);
+            outputSampleTypeField.setAccessible(true);
+
+            // channel count, sample rate, sample type
+            int key_channel_count = 0, key_sample_rate = 1, key_sample_type = 2, key_sample_type_result = 3;
+            int[][] tests = new int[][] {
+                    { 47, 42000, AudioFormat.ENCODING_PCM_8BIT, SampleType.UNSIGNED_8_BIT },
+                    { 2, 16000, AudioFormat.ENCODING_PCM_16BIT, SampleType.SIGNED_16_BIT },
+                    { 1, 22050, AudioFormat.ENCODING_PCM_FLOAT, SampleType.FLOAT },
+                    { 3, 48000, AudioFormat.ENCODING_INVALID, SampleType.SIGNED_16_BIT },
+            };
+
+            for (int[] test : tests) {
+                int channel_count = test[key_channel_count];
+                int sample_rate = test[key_sample_rate];
+                int sample_type = test[key_sample_type];
+                int sample_type_result = test[key_sample_type_result];
+
+                MediaFormat format = new MediaFormat();
+
+                format.setInteger(MediaFormat.KEY_CHANNEL_COUNT, channel_count);
+                format.setInteger(MediaFormat.KEY_SAMPLE_RATE, sample_rate);
+                format.setInteger(MediaFormat.KEY_PCM_ENCODING, sample_type);
+
+                onOutputFormatChangedMethod.invoke(mockDecoder, format);
+
+                assertEquals(channel_count, outputChannelCountField.getInt(mockDecoder));
+                assertEquals(sample_rate, outputSampleRateField.getInt(mockDecoder));
+                assertEquals(sample_type_result, outputSampleTypeField.getInt(mockDecoder));
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail();
+        }
+    }
+
+    private Method getSampleAtTargetMethod() {
+        Method method = null;
+        try {
+            method = BaseAudioDecoder.class.getDeclaredMethod("sampleAtTargetTime",
+                    double.class, SampleBuffer.class, double.class, double.class, double.class);
+            method.setAccessible(true);
+        } catch (NoSuchMethodException e) {
+            e.printStackTrace();
+            fail();
+        }
+        return method;
+    }
+
+    private BaseAudioDecoder createMockBaseAudioDecoder() {
+        AudioDecoderListener listener = new AudioDecoderListener() {

```

Comment:
Can use mockito to mock,
`AudioDecoderListener listener = mock(AudioDecoderListener.class);`